public/modules/custom/grants_handler/src/Form/MessageForm.php

```diff
@@ -164,9 +165,76 @@ public function buildForm(array $form, FormStateInterface $form_state, WebformSu
       '#value' => $this->t('Send'),
     ];
 
+    $form_state->setStorage($storage);
+
     return $form;
   }
 
+  /**
+   * Validate & upload file attachment.
+   *
+   * @param array $element
+   *   Element tobe validated.
+   * @param \Drupal\Core\Form\FormStateInterface $formState
+   *   Form state.
+   * @param array $form
+   *   The form.
+   *
+   * @throws \Drupal\helfi_atv\AtvDocumentNotFoundException
+   * @throws \Drupal\helfi_atv\AtvFailedToConnectException
+   */
+  public static function validateUpload(array &$element, FormStateInterface $formState, array &$form) {
+
+    $triggeringElement = $formState->getTriggeringElement();
+
+    if (str_contains($triggeringElement["#name"], 'messageAttachment_upload_button')) {
```

Comment:
Tyyliasioista keskusteltiinkin itse suosisin alla olevaa muotoa mutta en aloita asiasta tappelua jos et tee muutoksia.

```suggestion
    if (!str_contains($triggeringElement["#name"], 'messageAttachment_upload_button')) {
      return;
    }
```

Reply:
Ei mulla oo väliä. Toi on edelleen musta huono tapa tehdä toi, mut jos sä ja Sonar ootte toista mieltä ni mitäpä mä siihen inttämään.